service/history/queues/reader_group.go

```diff
@@ -111,7 +126,18 @@ func (g *ReaderGroup) NewReader(readerID int32, slices ...Slice) Reader {
 		panic(fmt.Sprintf("reader with ID %v already exists", readerID))
 	}
 
+	// TODO: return error if reader group already stopped
+
 	g.readerMap[readerID] = reader
+	err := g.executionManager.RegisterHistoryTaskReader(context.Background(), &persistence.RegisterHistoryTaskReaderRequest{
+		ShardID:      g.shardID,
+		TaskCategory: g.category,
+		ReaderID:     readerID,
+	})
+	if err != nil {
```

Comment:
Could it return transient errors?

Reply:
Right now no, since we are not talking to a remote service. Only configuration error could lead to this. 
And I will send out a follow up PR to not panic here and instead cancel the queue action that wants to create this new reader.